app/utils/forms.tsx

```diff
@@ -98,25 +99,42 @@ export function CheckboxField({
 	errors,
 }: {
 	labelProps: Omit<JSX.IntrinsicElements['label'], 'className'>
-	buttonProps: Omit<
-		React.ComponentPropsWithoutRef<typeof Checkbox.Root>,
-		'type' | 'className'
-	> & {
-		type?: string
-	}
+	buttonProps: Omit<JSX.IntrinsicElements['input'], 'className'>
 	errors?: ListOfErrors
 }) {
 	const fallbackId = useId()
 	const id = buttonProps.id ?? buttonProps.name ?? fallbackId
 	const errorId = errors?.length ? `${id}-error` : undefined
+	// To emulate naitve input events that Conform listen to:
+	// See https://conform.guide/integrations
+	const [ref, control] = useInputEvent()
+	const buttonRef = useRef<HTMLButtonElement>(null)
 	return (
 		<div className={styles.checkboxField}>
+			<input
```

Comment:
I'm concerned about this. Won't this mean we're rendering two <input /> elements per checkbox?

Reply:
Yes. But only this checkbox element will have the name attribute set. The one rendered by Radix will have no name and will be ignored for FormData.

Reply:
Should we make this a part of a generic checkbox component so people don't have to do this song and dance every time they need a checkbox?

Reply:
Sure. We are now modifying the `<CheckboxField />` component so people can reuse it. Or do you mean something else?

Reply:
Yep, that's what I'm thinking 👍